### ISSUE TITLE
可以训练，但是使用test_on_single_image.py测试训练好的模型没有bboxs，scores,

### ISSUE THREAD

**Issue by pursuityu** [calmisential/YOLOv3_TensorFlow2#2]
然后我又用官网提供的yolov3_coco.ckpt测试你的代码test_on_single_image.py，发现还是没有bbox，感觉是不是代码有问题，但是又没有报错。你测试有结果吗

**Comment by calmiLovesAI** [comment]
我测试的时候是正常的，因为使用test_on_single_image.py和在训练过程中开启测试单张图片用的是相同的代码。你在官网上下载的yolov3_coco.ckpt可能并不适用于这个项目，两者网络结构里面的细节不一定相同。

**Comment by herok94** [comment]
训练时候loss只有1.0左右，但是使用test_on_single_image.py没有输出，训练集有2400个样本，两类，只有正样本没有负样本

**Comment by pursuityu** [comment]
> 训练时候loss只有1.0左右，但是使用test_on_single_image.py没有输出，训练集有2400个样本，两类，只有正样本没有负样本

我也不知道怎么回事，我换了代码。

**Comment by calmiLovesAI** [comment]
> 
> 
> 训练时候loss只有1.0左右，但是使用test_on_single_image.py没有输出，训练集有2400个样本，两类，只有正样本没有负样本

你可以尝试增加训练轮数

**Comment by PEPSIOFME** [comment]
我也训练完后预测没有输出结果，老哥很奇怪

**Comment by nanshenwei** [comment]
我也没有/(ㄒoㄒ)/~~

**Comment by nanshenwei** [comment]
> 我也没有/(ㄒoㄒ)/~~

再训练很多轮之后有结果了，不过结果不太好，换个图又没结果了，估计是需要训练再就一点 @pursuityu

**Comment by ZhouKai90** [comment]
我训练多轮之后，单张图片做测试，有框，但是是乱的，我发现bounding_box.py第23行附近
area = h * w
pred = tf.reshape(feature_map, shape=(-1, ANCHOR_NUM_EACH_SCALE * area, CATEGORY_NUM + 5))
pred = tf.nn.sigmoid(pred)  #?????
tx_ty, tw_th, confidence, class_prob = tf.split(pred, num_or_size_splits=[2, 2, 1, CATEGORY_NUM], axis=-1) 
这个地方对所有的结果都进行了sigmoid计算，我理解的是只用对其中的class_prob进行sigmod，而不应该对tx_ty, tw_th进行变换，是不是这里导致了坐标的不准确@calmisential

**Comment by holdrain** [comment]
调了调终于能训练了，不过缺点就是没有预训练的权重，emmmm

**Comment by holdrain** [comment]
用笔记本吃不消
